### PR TITLE
api redux setup

### PR DESCRIPTION
**Pull request by pablo-clueless**
- Added endpoints CRUD operations in redux
- Changed routing for developers
- Made changes to the general tab

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -81,14 +81,12 @@ const App: React.FC = () => {
             <Route element={<PrivateRoutes />}>
               <Route path="/user/:id" element={<UserProfile />} />
               <Route path="/developers/dashboard" element={<DevDashboard />} />
-              <Route path="/developers" element={<DeveloperApiPage />} />
+              <Route path="/developers/api/:id" element={<DeveloperApiPage />} />
```
Comment (PhilemonBrain) [inline src/App.tsx]: change to /developer/api/:id

File: src/App.tsx [hunk 2]
```diff
@@ -81,14 +81,12 @@ const App: React.FC = () => {
             <Route element={<PrivateRoutes />}>
               <Route path="/user/:id" element={<UserProfile />} />
               <Route path="/developers/dashboard" element={<DevDashboard />} />
```
Comment (PhilemonBrain) [inline src/App.tsx]: change to /developer/dashbooard

File: src/components/EndpointTable.tsx [hunk 3]
```diff
@@ -7,21 +7,38 @@ import TableHead from '@mui/material/TableHead';
...
-    console.log(`editing route with id ${id}`)
+  const save = (id: string | undefined) => {
+    const payload = {id, name, method, route}
+    dispatch(editEndpoint(payload))
+    setIsEditing(null)
   }
   
-  const deleteRoute = (id: string) => {
-    console.log(`deleting route with id ${id}`)
+  const deleteRoute = (id: string | undefined) => {
```
Comment (PhilemonBrain) [inline src/components/EndpointTable.tsx:40]: you need to make a request to the BE to delete this. once u confirm the deletion, then call the dispatch.

File: src/components/EndpointTable.tsx [hunk 4]
```diff
@@ -7,21 +7,38 @@ import TableHead from '@mui/material/TableHead';
...
+  const dispatch = useAppDispatch()
   const classes = useStyles()
+  
+  const openEditing = (index: number) => {
+    setIsEditing(index)
+  }
 
-  const editRoute = (id: string) => {
-    console.log(`editing route with id ${id}`)
+  const save = (id: string | undefined) => {
```
Comment (PhilemonBrain) [inline src/components/EndpointTable.tsx:34]: you need to make a request to the BE to edit this. once u confirm the edit, then call the dispatch.

File: src/components/EndpointTab.tsx [hunk 5]
```diff
@@ -3,28 +3,28 @@ import { Paper, Stack, Typography } from "@mui/material";
...
-    const addEndpoint = (e: FormEvent) => {
+    const handleSubmit = (e: FormEvent) => {
         e.preventDefault()
 
         if(!name || !route) return toast.error("Please add a name and route")
         const payload = { id: Math.random().toFixed(10).toString(), name, route, method }
-        // console.log(payload)
         // * there will be a post and background get request whenever a new endpoint is added
-        mockEndpoint.unshift(payload)
+        dispatch(addEndpoint(payload))
```
Comment (PhilemonBrain) [inline src/components/EndpointTab.tsx]: we need to make a post request and confirm the request before calling the dispatch

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

File: src/components/ApiPageLayout/index.tsx [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
+
+interface Props { id: string | undefined }
+
+const ApiPageLayout:React.FC<Props> = ({id}) => {
     const classes = useStyles()
   return (
     <div className={classes.layout}>
        <Sidebar />
-       <Navbar />
+       <Navbar id={id} />
```
Comment (taiwoAkindele) [inline src/components/ApiPageLayout/index.tsx:14]: Why are you passing an id ?

**Pull request merged**